### PR TITLE
Return whole books instead of references on recomendation views

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: catalog/requirements.txt [hunk 1]
```diff
@@ -8,3 +8,4 @@ psycopg2-binary==2.8.6
 qservices-library==1.1
 wheel==0.35.1
 whitenoise==5.2.0
+qservices-library
```
Comment (genismoreno) [inline catalog/requirements.txt]: duplication!! Same in line 8!

File: docker-compose.yaml [hunk 2]
```diff
@@ -58,8 +58,8 @@ services:
 
   recomendator:
     build:
-      context: .
-      dockerfile: recomendator/Dockerfile
+      context: recomendator/
```
Comment (genismoreno) [inline docker-compose.yaml:63]: why is this appearing as a change if it's already in the splitting-the-monlith branch?

File: docker-compose.yaml [hunk 3]
```diff
@@ -70,7 +70,7 @@ services:
       - catalog
     volumes:
       - ./recomendator:/opt/recomendator
-      - ./common:/opt/recomendator/common
+      - ./common:/usr/local/lib/python3.8/site-packages/common
```
Comment (genismoreno) [inline docker-compose.yaml]: same here

File: recomendator/Dockerfile [hunk 4]
```diff
@@ -1,10 +1,9 @@
 FROM python:3.8
 
-COPY recomendator/requirements.txt /
+COPY requirements.txt /
```
Comment (genismoreno) [inline recomendator/Dockerfile]: same

File: recomendator/Dockerfile [hunk 5]
```diff
@@ -1,10 +1,9 @@
...
 RUN pip install -r requirements.txt
 
-COPY recomendator /opt/recomendator
-COPY common /opt/recomendator/common
+COPY . /opt/recomendator
```
Comment (genismoreno) [inline recomendator/Dockerfile]: same

File: api_gateway/views/recomendation.py [hunk 6]
```diff
@@ -1,20 +1,24 @@
 import os
 
 import grpc
-from apiflask import APIBlueprint, input, output, doc
-from flask import Blueprint, current_app
+from apiflask import APIBlueprint, output, doc
+from flask import current_app
```
Comment (genismoreno) [inline api_gateway/views/recomendation.py:5]: isn't this already in the splitting-the-monolith branch???

File: api_gateway/views/recomendation.py [hunk 7]
```diff
@@ -25,9 +29,13 @@
 @returns_json
 def get_list(user_id: int):
     try:
-        response = GRPC_STUB.List(recomendations_pb2.RecomendationsRequest(
+        book_ids = GRPC_RECOMENDATOR_STUB.List(recomendations_pb2.RecomendationsRequest(
             id=user_id
         ))
+        response = [
+            GRPC_CATALOG_STUB.Retrieve(book_pb2.BookRetrieveRequest(id=book_id.id))
```
Comment (genismoreno) [inline api_gateway/views/recomendation.py:36]: Is this better than making only one grpc query with all the ids?

**Review by genismoreno — CHANGES_REQUESTED** [review]

File: docker-compose.yaml (hunk 2 above)
Comment (ghost) [reply docker-compose.yaml:63]: The change was not applied in this branch cause git things XD

File: api_gateway/views/recomendation.py (hunk 7 above)
Comment (ghost) [reply api_gateway/views/recomendation.py:36]: Is that possible?  I only see request for retrieving all books or just one

Comment (genismoreno) [reply api_gateway/views/recomendation.py:36]: Well, we could add it, I'm not saying that it's already implemented. But if we are running out of time it's ok

Comment (ghost) [reply api_gateway/views/recomendation.py:36]: I think I could implement it today, lets talk it on the daily

**Pull request closed without merging**